src/data/ores/index.js

```diff
@@ -1,5 +1,24 @@
-export { bronzeOre } from './bronzeOre'
-export { goldOre } from './goldOre'
-export { ironOre } from './ironOre'
-export { silverOre } from './silverOre'
-export { stone } from './stone'
+import { bronzeOre } from './bronzeOre'
+import { goldOre } from './goldOre'
+import { ironOre } from './ironOre'
+import { silverOre } from './silverOre'
+import { stone } from './stone'
+
+export { bronzeOre, goldOre, ironOre, silverOre, stone }
+
+const ores = [bronzeOre, goldOre, ironOre, silverOre, stone].sort(
+  o => o.spawnChance
+)
+
+export function oreSpawner(value) {
+  var oreToSpawn = null
+
+  for (let ore of ores) {
+    if (value < ore.spawnChance) {
+      oreToSpawn = ore
+      break
+    }
+  }
+
+  return oreToSpawn
```

Comment:
@jeremyckahn i would like your feedback on this approach before i go too far. the way i have it, one random number is generated and then compared against all the ore spawn chances so order is important (lowest chance to highest). i rolled that up into this `oreSpawner` method, and just put it here to test out. does this seem like the right approach?

Reply:
As long as the implementation details are sufficiently abstracted away (which they seem to be), then it doesn't matter _too_ much how this logic works, so long as it behaves the way we want.

I'm hesitant to pick the code apart too much now because I think things will become much more intuitive and clear once we play test this a bit. I also don't have a great alternative implementation suggestion to offer at the moment. 😅

I think this is off to a good start, so let's get this minimally working and then iterate and polish it up once we have game behavior that feels right!

One thing I like about this is that since you're passing in the random value, this will be easy to test without mocking! 👍

Reply:
i ended up moving that random value bit because it ended up using more than 1 random number, but open to suggestions if you have them